ssl/statem/statem_srvr.c

```diff
@@ -3084,7 +3084,7 @@ static int tls_process_cke_dhe(SSL_CONNECTION *s, PACKET *pkt)
     }
 
     if (!EVP_PKEY_set1_encoded_public_key(ckey, data, i)) {
-        SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
+        SSLfatal(s, SSL_AD_ILLEGAL_PARAMETER, ERR_R_DH_LIB);
```

Comment:
Please use SSL_R_BAD_KEY_SHARE instead of ERR_R_DH_LIB and same in the ERR_R_EC_LIB case below.